ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -510,10 +524,25 @@ mkApps kernel Tracers {..} Codecs {..} genChainSyncTimeout Handlers {..} =
       -> remotePeer
       -> Channel m bKA
       -> m ((), Maybe bKA)
-    aKeepAliveClient _version _them _channel = do
+    aKeepAliveClient version them channel = do
       labelThisThread "KeepAliveClient"
-      -- TODO implement the client
-      forever (threadDelay 1000) >> return ((), Nothing)
+      let kacApp = case version of
+                        -- Version 1 doesn't support keep alive protocol but Blockfetch
+                        -- still requires a PeerGSV per peer.
+                        NodeToNodeV_1 -> \_ -> forever (threadDelay 1000) >> return ((), Nothing)
+                        NodeToNodeV_2 -> \_ -> forever (threadDelay 1000) >> return ((), Nothing)
+                        _             -> \dqCtx -> do
+                          runPeerWithLimits
+                            nullTracer
+                            cKeepAliveCodec
+                            (byteLimitsKeepAlive (const 0)) -- TODO: Real Bytelimits, see #1727
+                            timeLimitsKeepAlive
+                            channel
+                            $ keepAliveClientPeer
+                            $ hKeepAliveClient version (neverStop (Proxy :: Proxy m)) them dqCtx
+                                (KeepAliveInterval 10)
```

Comment:
Nitpicking, we prefer to indent with 2 spaces from `case`, not 5.

Reply:
I think that `case` is a special case. But I've no desire to fight an indentation war in consensus.

Reply:
Everybody has his/her own style, the style we follow in consensus is not my own style either (I prefer 4 spaces everywhere, actually).